package/ios/RNSkia-iOS/ViewScreenshotService.mm

```diff
@@ -0,0 +1,50 @@
+#import "ViewScreenshotService.h"
+#import <QuartzCore/QuartzCore.h>
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+
+#include "SkData.h"
+
+#pragma clang diagnostic pop
+
+@implementation ViewScreenshotService {
+  RCTUIManager *_uiManager;
+}
+
+- (instancetype)initWithUiManager:(RCTUIManager *)uiManager {
+  if (self = [super init]) {
+    _uiManager = uiManager;
+  }
+  return self;
+}
+
+- (sk_sp<SkImage>)screenshotOfViewWithTag:(NSNumber *)viewTag {
+  // Find view corresponding to the tag
+  auto view = [_uiManager viewForReactTag:viewTag];
+  if (view == NULL) {
+    RCTFatal(RCTErrorWithMessage(@"Could not find view with tag"));
+  }
+
+  // Get size
+  auto size = view.frame.size;
+
+  // Setup contextr
+  UIGraphicsBeginImageContextWithOptions(size, NO, 0);
+
+  // Render to context
+  //[view.layer renderInContext:UIGraphicsGetCurrentContext()];
+  [view drawViewHierarchyInRect:(CGRect){CGPointZero, size}
+             afterScreenUpdates:YES];
+
+  // Convert to image and release context
+  UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
+  UIGraphicsEndImageContext();
+
```

Comment:
We should be able to create an SkImage from an UIImage. This would enable us to avoid unnecessary PNG encoding.

Reply:
The following seems to work nicely:
```suggestion
-  NSData *data = UIImagePNGRepresentation(image);
-  sk_sp<SkData> skData = SkData::MakeWithCopy(data.bytes, data.length);
-  return SkImage::MakeFromEncoded(skData);
+  CGImageRef cgImage = image.CGImage;
+  size_t width = CGImageGetWidth(cgImage);
+  size_t height = CGImageGetHeight(cgImage);
+  size_t bitsPerComponent = 8;
+  size_t bytesPerPixel = 4;
+  size_t bytesPerRow = bytesPerPixel * width;
+  size_t bufferSize = bytesPerRow * height;
+
+  NSMutableData *data = [NSMutableData dataWithLength:bufferSize];
+  CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
+
+  CGContextRef bitmapContext = CGBitmapContextCreate(
+      data.mutableBytes, width, height, bitsPerComponent, bytesPerRow,
+      colorSpace, kCGImageAlphaPremultipliedLast | kCGBitmapByteOrder32Big);
+
+  CGContextDrawImage(bitmapContext, CGRectMake(0, 0, width, height), cgImage);
+
+  CGColorSpaceRelease(colorSpace);
+  CGContextRelease(bitmapContext);
+
+  SkImageInfo info = SkImageInfo::Make(width, height, kRGBA_8888_SkColorType,
+                                       kPremul_SkAlphaType);
+  sk_sp<SkData> skData = SkData::MakeFromMalloc(data.mutableBytes, bufferSize);
+  sk_sp<SkImage> skImage = SkImage::MakeRasterData(info, skData, bytesPerRow);
+
+  return skImage;
```

Reply:
Nice one :) Have investigated it and found a little issue with memory (where SkImage frees SkData and then SkData frees its buffer - but the buffer here is owned by NSData so it crashes). Made a version that doesn't copy data, just wraps the UIImage data into a SkData - seems to work really nice.